### PR TITLE
Skip write `prewrite_lock` in flashback locks

### LINKED ISSUE #13958: PITR: data inconsistency after PITR with log containing flashback data.

### PR DESCRIPTION
**Pull request by HuSharp**
Signed-off-by: husharp <jinhao.hu@pingcap.com>

### What is changed and how it works?

Issue Number: Close #13958

What's Changed:

```commit-message
Since the rollback ts for flashback are derived from the ts of the lock, we wrote prewrite lock which start_ts is flashback in Prewrite Phase. 
So the Prewrite lock we wrote in the flashback was rollbacked when we retry prepare. 
This introduces the case mentioned in https://github.com/tikv/tikv/issues/13958
The solution is: if such a lock exists, skip it and go to the Commit Phase.
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by HuSharp** [comment]
cc @JmPotato @joccau

File: tests/integrations/server/kv_service.rs [hunk 1]
```diff
@@ -810,6 +810,34 @@ fn test_mvcc_flashback_unprepared() {
     assert_eq!(get_resp.value, b"".to_vec());
 }
 
+#[test]
+fn test_mvcc_flashback_retry_prepare() {
```
Comment (JmPotato) [inline tests/integrations/server/kv_service.rs:814]: I'd prefer putting this test into the `src/storage/mod.rs/tests`.

**Review by JmPotato — APPROVED** [review]

**Comment by sticnarf** [comment]
@joccau Can you take a look at the PR?

**Review by joccau — APPROVED** [review]
LGTM

**Review by sticnarf — APPROVED** [review]

**Comment by JmPotato** [comment]
/cherry-pick-to-release-6.5

**Pull request merged**